packages/libs/components/src/map/MapVEuMap.tsx

```diff
@@ -476,3 +442,142 @@ function MapScrollWheelZoom(props: MapScrollWheelZoomProps) {
 
   return null;
 }
+
+// custom zoom control
+interface CustomZoomControlProps {
+  defaultViewport?: Viewport;
+}
+
+function CustomZoomControl(props: CustomZoomControlProps) {
+  const map = useMap();
+
+  // zoom in
+  const zoomIn = (e: React.SyntheticEvent) => {
+    e.preventDefault();
+    // map.doubleClickZoom.disable();
+    map.setZoom(map.getZoom() + 1);
+  };
+
+  // zoom out
+  const zoomOut = (e: React.SyntheticEvent) => {
+    e.preventDefault();
+    map.setZoom(map.getZoom() - 1);
+  };
+
+  // zoom to whole world
+  const zoomToWholeWorld = (e: React.SyntheticEvent) => {
+    e.preventDefault();
+    // with zoom level 1, then center: [1,1] works
+    map.setView([1, 1], 1);
```

Comment:
(1,1) seems a bit arbitrary, and I think the global view should be the same as the default viewport.

Could we have something like this please?

```
const epsilon = 0.25; // or whatever smallest value works reliably (can we understand why very small values don't work?)
map.setView(props.defaultViewport.center.map((coord) => coord + epsilon), props.defaultViewport.zoom);
```

Reply:
I used the approach, epsilon. Although it looks okay by using map function to sum coord and epsilon, however, it somehow causes type error. Thus, I simply utilized [center[0]+epsilon, center[1]+epsilon]

As I mentioned before, it seems to me that there exists a coordinate resolution per zoom level. For zoom level 1, it looks like 0.703... Thus, to distinguish zoom-to-whole-world from zoom-to-data, epsilon should be larger than 0.703... For this reason, I just used epsilon = 1 considering some margin of error.

Reply:
Ah, thank you - I didn't realise it was zoom-level related. Apologies if you explained already...

That 0.703 looks like it could be `1/sqrt(2)`